src/main/java/edu/harvard/iq/dataverse/util/DateUtil.java

```diff
@@ -0,0 +1,77 @@
+package edu.harvard.iq.dataverse.util;
+
+import edu.harvard.iq.dataverse.DataverseLocaleBean;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+/**
+ *
+ * @author jchengan
+ */
+public class DateUtil {
+
+    private static final Logger logger = Logger.getLogger(DateUtil.class.getCanonicalName());
+
+
+    public static String formatDate(Date dateToformat) {
+        try
+        {
+            DataverseLocaleBean d = new DataverseLocaleBean();
+            Locale bundle_locale = new Locale(d.getLocaleCode());
+
+            String formattedDate = "";
+
+            if(bundle_locale.getLanguage().contains("fr"))
```

Comment:
With over half the world's population (*Wikipedia: https://en.wikipedia.org/wiki/Date_format_by_country) using the "dd MMM yyyy" format, there must be a more scalable solution than hardcoding "fr" here. @scolapasta might have some more detailed insight to share.

Reply:
So 1) we should use DateFormatter rather than SimpleDateFormat. If we use it like below, then we don't need to have the conditional:
dateFormatter = DateFormat.getDateInstance(DateFormat.DEFAULT, currentLocale);
today = new Date();
dateOut = dateFormatter.format(today);

We could use either DEFAULT or MEDIUM. (for example DEFAULT for German would produce: 30.06.2009). If we want to have the text, I think we need to use MEDIUM in that case.

(also, the second method should call the first; i.e. convert the String to date, then pass that to the first method)